sdk/iot/hub/samples/twin/src/iot_hub_twin_example.c

```diff
@@ -0,0 +1,447 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(default : 4201)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <az_iot_hub_client.h>
+#include <az_result.h>
+#include <az_span.h>
+#include <az_json.h>
+
+// TODO: #564 - Remove the use of the _az_cfh.h header in samples.
+//              Note: this is required to work-around MQTTClient.h as well as az_span init issues.
+#include <_az_cfg.h>
+
+// Device ID
+#define DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// IoT Hub FQDN
+#define IOT_HUB_FQDN "AZ_IOT_HUB_FQDN"
+
+// Device information
+#define REGISTRATION_ID_ENV "AZ_IOT_REGISTRATION_ID"
+
+// AZ_IOT_DEVICE_X509_CERT_PEM_FILE is the path to a PEM file containing the device certificate and
+// key as well as any intermediate certificates chaining to an uploaded group certificate.
+#define DEVICE_X509_CERT_PEM_FILE "AZ_IOT_DEVICE_X509_CERT_PEM_FILE"
+
+// AZ_IOT_DEVICE_X509_TRUST_PEM_FILE is the path to a PEM file containing the server trusted CA
+// This is usually not needed on Linux or Mac but needs to be set on Windows.
+#define DEVICE_X509_TRUST_PEM_FILE "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE"
+
+static char device_id[64];
+static char iot_hub_fqdn[128];
+static char x509_cert_pem_file[512];
+static char x509_trust_pem_file[256];
+
+static char mqtt_endpoint[128];
+static az_span mqtt_url_prefix = AZ_SPAN_LITERAL_FROM_STR("ssl://");
+static az_span mqtt_url_suffix = AZ_SPAN_LITERAL_FROM_STR(":8883");
+
+static char get_twin_topic[128];
+static az_span get_twin_topic_request_id = AZ_SPAN_LITERAL_FROM_STR("get_twin");
```

Comment:
To set a good example, I think this should be as short as possible (starting with one byte). The only requirement from hub is to have this unique.

Reply:
The idea I had here was to easily be able to associate a returned response to the request which originated. This way if someone gets an error return, they can tell if it was from a `get_twin` or `reported_properties`